### ISSUE TITLE
add support for m4 version 1.4.19.0.4.gcd7f

### ISSUE THREAD

**Issue by viy2** [PerlAlien/Alien-m4#18]
There is a version of m4 with characters which is not recognized.
$ rpm -q m4
m4-1.4.19.0.4.gcd7f-alt1.x86_64

$ m4 --version
m4 (GNU M4) 1.4.19.0.4.gcd7f
Copyright (C) 2021 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by Rene' Seindal.

see
http://ftp.altlinux.org/pub/distributions/ALTLinux/Sisyphus/files/SRPMS/m4-1.4.19.0.4.gcd7f-alt1.src.rpm

proposed patch:
```diff
--- Alien-m4-0.20/alienfile	2021-06-22 19:36:22.354808420 +0000
+++ Alien-m4-0.20/alienfile	2021-06-22 19:37:41.542008089 +0000
@@ -15,7 +15,7 @@
   {
     my @stdout;
     capture { @stdout = `$command --version` };
-    if(defined $stdout[0] && $stdout[0] =~ /^(GNU |)m4/i && $stdout[0] =~ /([0-9\.]+)$/)
+    if(defined $stdout[0] && $stdout[0] =~ /^(GNU |)m4/i && $stdout[0] =~ /([0-9\.]+)(?:\.\w[\w\d\.]*)?$/)
     {
       my $version = $1;
       my @version = split /\./, $version;
```

**Comment by plicease** [comment]
We could probably also just remove the `$` anchor in the regexp?

**Comment by plicease** [comment]
Can you try this patch:
https://github.com/PerlAlien/Alien-m4/pull/19/commits/0100ed1b919a6dca37aa394c013fbac89c36f76f

**Comment by plicease** [comment]
Can you try this patch:

**Comment by plicease** [comment]
https://github.com/PerlAlien/Alien-m4/commit/05da7fc2e534d7a379489993ffd54088acf621ab.diff

I think GNU m4 always has three numbers in the triplet

**Comment by viy2** [comment]
your patch works for that m4, please, do as you find proper.

**Comment by plicease** [comment]
Great.  This should be addressed in 0.21.
